clang/include/clang/Sema/Sema.h

```diff
@@ -14512,20 +14512,6 @@ class Sema final {
       DeviceDiagnosticReason Reason = DeviceDiagnosticReason::Sycl |
                                       DeviceDiagnosticReason::Esimd);
 
-  /// Check whether we're allowed to call Callee from the current context.
-  ///
-  /// - If the call is never allowed in a semantically-correct program
-  ///   emits an error and returns false.
-  ///
-  /// - If the call is allowed in semantically-correct programs, but only if
-  ///   it's never codegen'ed, creates a deferred diagnostic to be emitted if
-  ///   and when the caller is codegen'ed, and returns true.
-  ///
-  /// - Otherwise, returns true without emitting any diagnostics.
-  ///
-  /// Adds Callee to DeviceCallGraph if we don't know if its caller will be
-  /// codegen'ed yet.
-  bool checkSYCLDeviceFunction(SourceLocation Loc, FunctionDecl *Callee);
```

Comment:
This function is coming from upstream, I don't think deleting it from intel/llvm is a good idea.

Reply:
It is actually quite dump there and doesn't do anything, I'm not sure how I ended up committing it when implemented https://reviews.llvm.org/D74387 . In any way, since this function is coming from upstream, it should be removed from there, not from intel/llvm.

Reply:
It doesn't seem to be doing anything in community clang either - https://clang.llvm.org/doxygen/SemaSYCL_8cpp_source.html. Function will always return true. I'm checking out a new community workspace now since my old workspace was wiped when servers were upgraded. I will make this change in community clang unless I see a reason for it in git history.

Reply:
Oh I didn't see your comment before I wrote mine. But yea I will go ahead and delete it in community then. Thanks for taking a look!